### ISSUE TITLE
0.8.14 failing to install from pypi

### ISSUE THREAD

**Issue by franzlang** [reflectometry/refl1d#104]
The latest version of refl1d on pypi.org (0.8.14) is failing to build on our systems (CentOS 7, python3.6). Version 0.8.12 works absolutely fine. I have attached the full output of calling `python3 -m pip install refl1d`. 
[refl1d_install.log](https://github.com/reflectometry/refl1d/files/5991097/refl1d_install.log)

**Comment by bmaranville** [comment]
c++ 11 support seems to be needed...  can you try adding an extra_compile_args flag to setup.py for the extension?

e.g.

ext = Extension('foo', sources=[....],
                libraries=[....], 
                extra_compile_args=['-std=c++11'],
                ....)

**Comment by acaruana2009** [comment]
Looks like this issue is still present. I have just tried to pip install into the user directory. I think it is an issue we have on IDAaaS with the gcc compliler.

To install as you are suggesting @bmaranville would we need to do this from source using the setup.py file?

Thanks

Andrew

**Comment by bmaranville** [comment]
This works on RHEL:
CFLAGS="-std=c++11" pip install --upgrade refl1d

**Comment by franzlang** [comment]
Setting CFLAGS as mentioned indeed resolved the issue for me as well on CentOS 7.

**Comment by pkienzle** [comment]
We should leave this open until we modify setup.py to include the necessary flags, or we modify the code so the flags are no longer necessary, or at a minimum we document the necessary flags in the install instructions.

**Comment by bmaranville** [comment]
I have a fix ready, but I don't want to push since @pkienzle might have a better solution that doesn't require c++11 compatibility
```patch
diff --git a/setup.py b/setup.py
index 1a5550a..6b99714 100755
--- a/setup.py
+++ b/setup.py
@@ -23,7 +23,12 @@ for line in open(os.path.join("refl1d", "__init__.py")):
         version = line.split('"')[1]
 #print("Version: "+version)
 
-extra_compile_args =  {'msvc': ['/EHsc']}
+# TODO: remove c++ version pinning when c++11 is most common g++ standard
+# (i.e. when RHEL7 is not commonly used anymore)
+extra_compile_args =  {
+    'msvc': ['/EHsc'],
+    'unix': ['-std=c++11'],
+}
 extra_link_args =  {}
```

**Comment by pkienzle** [comment]
Twiddling setup.py is a reasonable solution. We'll have to test whether it breaks OS/X builds.

The other option is to make specific macros for double and integer vectors so we don't have to rely on C++11 type introspection.

**Comment by bmaranville** [comment]
Indeed, instead of throwing a warning it throws an error on OS/X when one of the C files (not C++) is compiled, so the above patch won't work for building on OS/X.

**Comment by bmaranville** [comment]
The OSX problem goes away if all the C files are renamed to .cc ... is that an acceptable solution?

**Comment by pkienzle** [comment]
I suppose. We're going to replace this all with numba and/or opencl/cuda anyway at some point.

**Comment by bmaranville** [comment]
It's not pretty, but closed by 8c3b1f461471f15271523ed8c54a9270df426113
No CFLAGS needed for install on RHEL7.
